notes.md

```diff
@@ -0,0 +1,1900 @@
+</td>
+<td>
+
+```Python
+>>> import torch
+>>> from torch import nn
+>>> module = nn.Linear(5, 5)
+>>> i = torch.randn(2, 5, requires_grad=True)
+>>> module(i).sum().backward()
+>>> module.zero_grad()
+>>> module.weight.grad == None
+True
+>>> module.weight.grad.data
+AttributeError: 'NoneType' object has no attribute 'data'
+>>> module.weight.grad + 1.0
+TypeError: unsupported operand type(s) for +: 'NoneType' and 'float'
+```
+
+</td>
+</tr>
+</table>
+
+### **`torch.nn.utils.stateless.functional_call` now respects tied weights (#90477)**
+
+Assume a module has two tied weights, x and x_tied. Previously, invoking `functional_call(module, parameters_and_buffers, args, kwargs=None, *, strict=False)` with a parameter dictionary of only one of the tied weights would result in the other one(s) not being updated.
+
+We’ve changed the behavior so that providing one of the tied weights in the parameter dictionary will update all other tied weights. If you would like the behavior in previous versions of PyTorch, please set `tie_weights=False`.
+
+Please also see the related deprecation section "torch.nn.stateless.functional_call in favor of torch.func.functional_call".
+
+<table>
+<tr>
+<th>1.13</th>
+<th>2.0</th>
+</tr>
+<tr>
+<td>
+
+```Python
+>>> class Foo(nn.Module):
+...    def __init__(self):
+...        super().__init__()
+...        self.x = nn.Parameter(torch.zeros([]))
+...        self.x_tied = self.x
+...
+...    def forward(self, inp):
+...        return self.x + self.x_tied
+
+>>> foo = Foo()
+>>> params = {'x': torch.ones([])}
+>>> result = torch.nn.utils.stateless.functional_call(foo, params, torch.randn([]))
+>>> print(result)
+1.0
+```
+
+</td>
+<td>
+
+```Python
+>>> class Foo(nn.Module):
+...    def __init__(self):
+...        super().__init__()
+...        self.x = nn.Parameter(torch.zeros([]))
+...        self.x_tied = self.x
+...
+...    def forward(self, inp):
+...        return self.x + self.x_tied
+
+>>> foo = Foo()
+>>> params = {'x': torch.ones([])}
+>>> result = torch.nn.utils.stateless.functional_call(foo, params, torch.randn([]), tie_weights=False)
+>>> print(result)
+1.0
+```
+
+</td>
+</tr>
+</table>
+
+### **Require `return_complex` to be passed explicitly to `torch.stft` for real input (#86724)**
+
+`torch.stft` takes an optional return_complex parameter that indicates whether the output should be a floating point tensor or a complex tensor. `return_complex` previously defaulted to False for real input tensors. This PR removes the default and makes `return_complex` a required argument for real inputs. However, complex inputs will continue to default to `return_complex=True`.
+
+<table>
+<tr>
+<th>1.13</th>
+<th>2.0</th>
+</tr>
+<tr>
+<td>
+
+```Python
+>>> a = torch.rand(1024)
+>>> _ = torch.stft(a, n_fft=128)
+```
+
+</td>
+<td>
+
+```Python
+>>> t = torch.rand(1024)
+>>> _ = torch.stft(t, n_fft=128, return_complex=False)
+```
+
+</td>
+</tr>
+</table>
+
+### **Require inputs to `torch.istft` to be complex valued**
```

Comment:
nit: easier to see what's going on if you also include the like-for-like example (presumably with an error in 2.0).